src/lib/server/then_operator.hpp

```diff
@@ -1,5 +1,7 @@
 #pragma once
 
+#define BOOST_THREAD_PROVIDES_FUTURE
+#define BOOST_THREAD_PROVIDES_FUTURE_CONTINUATION
```

Comment:
I don't even fully understand what this changes, but it fixes the compilation problems with the new boost version. This whole file should soon be gone anyway.